SonarQube.MSBuild.Tasks/TaskUtilities.cs

```diff
@@ -0,0 +1,95 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TaskUtilities.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace SonarQube.MSBuild.Tasks
+{
+    /// <summary>
+    /// Helper methods used by multiple tasks
+    /// </summary>
+    public static class TaskUtilities
+    {
+        // Workaround for the file locking issue: retry after a short period.
+        public const int MaxConfigRetryPeriodInMilliseconds = 2500; // Maximum time to spend trying to access the config file
+
+        public const int DelayBetweenRetriesInMilliseconds = 499; // Period to wait between retries
+
+        #region Public methods
+
+        /// <summary>
+        /// Attempts to load the analysis config from the specified directory.
+        /// Will retry if the file is locked.
+        /// </summary>
+        /// <returns>The loaded configuration, or null if the file does not exist or could not be
+        /// loaded e.g. due to being locked</returns>
+        public static AnalysisConfig TryGetConfig(string configDir, ILogger logger)
```

Comment:
There are a few minor changes to the code that has been moved:
* the methods are now static so the required arguments have to be passed in
* the method now using a standard SonarQube Common.ILogger as that is easier to test. This means the caller has to use the existing MSBuildLoggerAdapter to convert between the MSBuild logging framework and ours.